### ISSUE TITLE
Preheating with Harbor 2.4 fails with "Error:Field validation for 'Tag' failed on the 'required' tag"

### ISSUE THREAD

**Issue by fbuchmeier-abi** [dragonflyoss/Dragonfly2#1362]
### Bug report:

I am trying to connect Harbor 2.4 (deployed with the Harbor Operator) to Dragonfly 2.0.3 (deployed with helm).

I have set up the connection in Harbor as specified here: https://goharbor.io/docs/2.4.0/administration/p2p-preheat/manage-preheat-providers/

According to the Dragonfly Preheat docs (https://d7y.io/docs/reference/preheat/api) a request should be made to the manager API like this:

```
curl --location --request POST 'http://localhost:8080/api/v1/jobs' \
--header 'Content-Type: application/json' \
--data-raw '{
    "type": "preheat",
    "args": {
        "type": "image",
        "url": "https://registry-1.docker.io/v2/library/redis/manifests/latest"
    }
}'
```

This is actually not working for me, as the required key `tag` is missing from the list of `args`. If I add the `tag` key to the `args`, the request to `/api/v1/jobs` succeeds. This can be seen in the `scheduler` log file and will result in the following error:

```
{
    "level": "error",
    "ts": "2022-06-03 13:44:38.069",
    "caller": "job/job.go:146",
    "msg": "url https://registry-1.docker.io/v2/library/redis/blobs/sha256:3b8feb83bfba22b33fa674aaab3126838387f64a8ebf86262a44c1f038b78f83 validate failed: Key: 'PreheatRequest.Tag' Error:Field validation for 'Tag' failed on the 'required' tag",
    "stacktrace": "d7y.io/dragonfly/v2/scheduler/job.(*job).preheat\n\t/go/src/d7y.io/dragonfly/v2/scheduler/job/job.go:146\nreflect.Value.call\n\t/usr/local/go/src/reflect/value.go:543\nreflect.Value.Call\n\t/usr/local/go/src/reflect/value.go:339\ngithub.com/RichardKnop/machinery/v1/tasks.(*Task).Call\n\t/go/pkg/mod/github.com/!richard!knop/machinery@v1.10.6/v1/tasks/task.go:142\ngithub.com/RichardKnop/machinery/v1.(*Worker).Process\n\t/go/pkg/mod/github.com/!richard!knop/machinery@v1.10.6/v1/worker.go:182\ngithub.com/RichardKnop/machinery/v1/brokers/redis.(*Broker).consumeOne\n\t/go/pkg/mod/github.com/!richard!knop/machinery@v1.10.6/v1/brokers/redis/redis.go:349\ngithub.com/RichardKnop/machinery/v1/brokers/redis.(*Broker).consume.func2\n\t/go/pkg/mod/github.com/!richard!knop/machinery@v1.10.6/v1/brokers/redis/redis.go:312"
}
```

I have traced the HTTP calls from Harbor and found out that Harbor is calling another endpoint `/preheats` instead of the jobs endpoint. The request body is also slightly different:

```
curl --location --request POST 'http://localhost:8080/preheats' \
--header 'Content-Type: application/json' \
--data-raw '{
    "type": "image",
    "url": "https://registry-1.docker.io/v2/library/redis/manifests/latest",
    "headers": {
        "Authorization": "Bearer ...
    },
    "image": "library/redis",
    "tag": "latest",
    "digest": "sha256:2c78c0ed0400eae8011663610e6f82828de64e91e2a5d4e501d53541df350c40"
}'
```

As one can see the call from Harbor to `/preheats` includes the `tag` key. However, this still returns the same error `Error:Field validation for 'Tag' failed on the 'required' tag` as above in the scheduler pod.

Looking at the code I can see that the struct CreateV1PreheatRequest in the `manager` does not include `Tag`: https://github.com/dragonflyoss/Dragonfly2/blob/main/manager/types/preheat.go#L19

The `job` struct PreheatRequest does include `Tag`: https://github.com/dragonflyoss/Dragonfly2/blob/main/internal/job/types.go

Tags are also not defined in the swagger file for the manager API: https://github.com/dragonflyoss/Dragonfly2/blob/main/api/manager/swagger.yaml#L520

As a workaround, I've tried to set the `Tag` as a required parameter for the `/preheats` call in the manager API. This did not cause the `tag` to be sent to the scheduler.

I have then tried to remove the `required` field from the scheduler, which caused the request to succeed. At the moment however I do not have any indication that the preheating was actually successful.

```
diff --git a/internal/job/types.go b/internal/job/types.go
index 45923f2d..f37a0bb5 100644
--- a/internal/job/types.go
+++ b/internal/job/types.go
@@ -18,7 +18,7 @@ package job
 
 type PreheatRequest struct {
        URL     string            `json:"url" validate:"required,url"`
-       Tag     string            `json:"tag" validate:"required"`
+       Tag     string            `json:"tag" validate:"omitempty"`
        Digest  string            `json:"digest" validate:"omitempty"`
        Filter  string            `json:"filter" validate:"omitempty"`
        Headers map[string]string `json:"headers" validate:"omitempty"`
diff --git a/manager/types/preheat.go b/manager/types/preheat.go
index 930de0c8..7dc52c36 100644
--- a/manager/types/preheat.go
+++ b/manager/types/preheat.go
@@ -19,6 +19,7 @@ package types
 type CreateV1PreheatRequest struct {
        Type    string            `json:"type" binding:"required,oneof=image file"`
        URL     string            `json:"url" binding:"required"`
+       TAG     string            `json:"tag" binding:"required"`
        Filter  string            `json:"filter" binding:"omitempty"`
        Headers map[string]string `json:"headers" binding:"omitempty"`
 }
```

### Expected behavior:

The `tag` key sent by Harbor to the `/preheats` endpoint should be passed downstream from the `manager` to the `scheduler` and preheat requests should thus succeed.

### How to reproduce it:

Install the official Helm Chart for Dragonfly:

``` 
chart: dragonfly
version: 0.6.0
```

Create a port-forwarding to the manager service:

```
kubectl port-forward svc/dragonfly-manager 8080:8080
```

Send a POST request to the same endpoint as Harbor does with the same parameters:

```
curl --location --request POST 'http://localhost:8080/preheats' \
--header 'Content-Type: application/json' \
--data-raw '{
    "type": "image",
    "url": "https://registry-1.docker.io/v2/library/redis/manifests/latest",
    "headers": {
        "Authorization": "Bearer ...
    },
    "image": "library/redis",
    "tag": "latest",
    "digest": "sha256:2c78c0ed0400eae8011663610e6f82828de64e91e2a5d4e501d53541df350c40"
}'
```

Then check the status of the job, it will have failed:

```
curl --request GET 'http://localhost:8080/api/v1/jobs/1'
```

To validate the reason for this failure, check the `scheduler` logs:

```
less /var/log/dragonfly/scheduler/core.log
```

### Environment:
- Dragonfly version: 2.0.3
- OS: Kubernetes, Amazon Linux 2
- Kernel (e.g. `uname -a`): 5.4.190-107.353.amzn2.x86_64
- Others: Harbor 2.4.2

I would be grateful for any tips on how to proceed :-)

Thanks,
Florian.

**Comment by LetFu** [comment]
I also had this problem, and i successfully preheat by specifying the image mediaType in the tag of --data-raw args. The command is like follows:
```bash
 curl --location --request POST 'http://101.42.243.65:30808/api/v1/jobs' \
--header 'Content-Type: application/json' \
--data-raw '{
    "type": "preheat",
    "args": {
        "type": "image",
        "url": "https://registry-1.docker.io/v2/library/redis/manifests/latest",
        "tag": "vnd.docker.distribution.manifest.v2+json"
    }
}'
```

**Comment by gaius-qi** [comment]
Fix #1363
